packages/@coorpacademy-components/src/molecule/video-player/index.js

```diff
@@ -5,33 +5,56 @@ import JWPlayer from './jwplayer';
 import Vimeo from './vimeo';
 import style from './style.css';
 
-const Player = props => {
-  const {mimeType, id, url, width = '100%', height = '400px', onPlay, onPause, onEnded} = props;
-  switch (mimeType) {
-    case 'application/vimeo':
-      return <Vimeo {...props} video={id} className={style.wrapper} />;
-
-    case 'application/youtube':
-    case 'application/kontiki':
-      return (
-        <VideoIframe
-          type={mimeType.split('application/')[1]}
-          id={id}
-          url={url}
-          width={width}
-          height={height}
-          frameBorder={0}
-          allowFullScreen
-          onPlay={onPlay}
-          onPause={onPause}
-          onEnded={onEnded}
-        />
-      );
-
-    case 'video/mp4':
-      return <JWPlayer {...props} />;
+class Player extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      played: false
+    };
+    this.handleOnPlay = this.handleOnPlay.bind(this);
   }
-};
+
+  handleOnPlay(e) {
+    if (this.state.played) {
+      this.props.onResume && this.props.onResume(e);
+    } else {
+      this.props.onPlay && this.props.onPlay(e);
+    }
+    this.setState({
+      played: true
+    });
+  }
+
+  render() {
+    const {mimeType, id, url, width = '100%', height = '400px', onPause, onEnded} = this.props;
+    switch (mimeType) {
+      case 'application/vimeo':
+        return (
+          <Vimeo {...this.props} video={id} className={style.wrapper} onPlay={this.handleOnPlay} />
+        );
+
+      case 'application/youtube':
+      case 'application/kontiki':
+        return (
+          <VideoIframe
+            type={mimeType.split('application/')[1]}
+            id={id}
+            url={url}
+            width={width}
+            height={height}
+            frameBorder={0}
+            allowFullScreen
+            onPlay={this.handleOnPlay}
+            onPause={onPause}
+            onEnded={onEnded}
+          />
+        );
+
+      case 'video/mp4':
+        return <JWPlayer {...this.props} />;
```

Comment:
du coup en utilisant directement l'api de jwplayer pour le onResume, tu ne set jamais ton state.played = true;
faudrait peut etre utiliser aussi `onPlay={this.handleOnPlay}` et `onResume={this.handleOnPlay}` ici non ?
peut etre un peu tordu pour pas gd chose.

Reply:
non, vu que le JWPlayer gère bien le onPlay et le onResume (c'st dans l'api) il n'y a pas besoin